### ISSUE TITLE
Supported OS platform name contains a space, causing a lot of CA1418 warnings

### ISSUE THREAD

**Issue by jnm2** [microsoft/win32metadata#627]
The VS error list and `dotnet build` show a bunch of warnings for `PInvoke.Msi.g.cs`, copies of:

> warning CA1418: The platform 'Windows ' is not a known platform name

Full repro (SDK version 6.0.100-preview.7.21379.14):

```
MsiGetComponentPath
MsiGetShortcutTarget
```

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Windows.CsWin32" Version="0.1.506-beta">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>

</Project>
```

**Comment by mikebattista** [comment]
@tannergooding do you know where these spaces might be getting encoded? I'm looking through this repo to understand where that might be happening and I see the assignment of these attributes in https://github.com/microsoft/win32metadata/blob/9d9ec4d349850c04f412b465b4c577234d9ab10c/generation/scraper/supportedOS.rsp#L6128 which is then passed to ClangSharpInvokeGenerator at https://github.com/microsoft/win32metadata/blob/9d9ec4d349850c04f412b465b4c577234d9ab10c/scripts/GenerateMetadataSourceForPartition.ps1#L140.

It doesn't look like the values we're passing in are encoded.

**Comment by mikebattista** [comment]
This might also be solved by #638 which proposes removing spaces.

**Comment by tannergooding** [comment]
@mikebattista, ClangSharp just parses the command line as given into key value pairs: https://github.com/microsoft/ClangSharp/blob/main/sources/ClangSharpPInvokeGenerator/Program.cs#L163 + https://github.com/microsoft/ClangSharp/blob/main/sources/ClangSharpPInvokeGenerator/Program.cs#L565-L589

Those get passed in, verbatim, to the configuration settings: https://github.com/microsoft/ClangSharp/blob/main/sources/ClangSharpPInvokeGenerator/Program.cs#L470 + https://github.com/microsoft/ClangSharp/blob/main/sources/ClangSharp.PInvokeGenerator/PInvokeGeneratorConfiguration.cs#L125

They are then consumed verbatim as well: https://github.com/microsoft/ClangSharp/blob/main/sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs#L1254-L1288

So, as far as I'm aware, there shouldn't be anything on the ClangSharp side that could cause a regular space to be replaced with a `nbsp` outside it actually being that way in the response file.

I do know that win32metadata does some of its own post-processing, however. Is it possible that its getting inserted there instead, possibly as part of some normalization or other cleanup that happens?

**Comment by tannergooding** [comment]
In particular, these Regex look suspicious: https://github.com/microsoft/win32metadata/blob/86cb5c17a97c20658d90403c0044a5c5c44844c0/scripts/CreateSupportedOSPlatformRsp.ps1#L39-L42

**Comment by mikebattista** [comment]
Ok. Thanks for the pointers.

**Comment by mikebattista** [comment]
Ok. I found the issue. The text in supportedOS.rsp has the special space character. I did previously replace all occurrences of "Windows 8" and "Windows Vista" with plain text before and rebuilt the winmd, but that had no effect. The missing piece was rescraping the partitions as well for all the affected APIs for the changes to take effect. I'm seeing the changes take effect now after rescraping.
